From 2ccea98374dac27446d74a7c66cd83a62b8cc21a Mon Sep 17 00:00:00 2001
From: Rafael Silva <rg-silva@criticalsoftware.com>
Date: Tue, 25 Jun 2024 10:55:56 +0100
Subject: [PATCH] cmake: add option to skip building tests

---
 CMakeLists.txt      | 1 +
 test/CMakeLists.txt | 4 ++++
 2 files changed, 5 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f6eb72dd..c79111b9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,6 +28,7 @@ set(CMAKE_CXX_STANDARD 17)
 
 option(COA_SETUP_CLANGFORMAT "Setups target to beautify the code with 'make clangformat'" OFF)
 option(COA_USE_QT6 "Compile with Qt6" OFF)
+option(COA_BUILD_TESTS "Build unit tests" ON)
 option(COA_BUILD_DOCUMENTATION "Build documentation" OFF)
 option(COA_NO_CODAC "Don't look for the presence of CODAC environment" OFF)
 
diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index abbc319e..5445dbe6 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -1,3 +1,7 @@
+if(NOT COA_BUILD_TESTS)
+  return()
+endif()
+
 enable_testing()
 
 find_package(GTest 1.10 QUIET CONFIG)
-- 
2.45.2

