diff --git a/src/tools/genVersionHeader.pl b/src/tools/genVersionHeader.pl
index bac9123..a195039 100644
--- a/src/tools/genVersionHeader.pl
+++ b/src/tools/genVersionHeader.pl
@@ -20,6 +20,9 @@ use POSIX qw(strftime);
 
 use strict;
 
+# Make sure that chomp removes all trailing newlines
+$/='';
+
 # RFC 8601 date+time w/ zone (eg "2014-08-29T09:42-0700")
 my $tfmt = '%Y-%m-%dT%H:%M';
 $tfmt .= '%z' unless $^O eq 'MSWin32'; # %z returns zone name on Windows
