From 71104a8cc5fee6a576857abe299a43fdd7f16aff Mon Sep 17 00:00:00 2001
From: Rafael Silva <rg-silva@criticalsoftware.com>
Date: Tue, 25 Jun 2024 10:39:16 +0100
Subject: [PATCH] cmake: add option to skip building tests

---
 CMakeLists.txt | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f63cde0..ed435cd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -29,6 +29,7 @@ set(CMAKE_CXX_STANDARD 17)
 option(COA_EXPORT_BUILD_TREE "Export build tree in /home/user/.cmake registry" OFF)
 option(COA_SETUP_CLANGFORMAT "Setups target to beautify the code with 'make clangformat'" OFF)
 option(COA_USE_QT6 "Compile with Qt6" OFF)
+option(COA_BUILD_TESTS "Build unit tests" ON)
 option(COA_BUILD_DOCUMENTATION "Build documentation" OFF)
 option(COA_NO_CODAC "Don't look for the presence of CODAC environment" OFF)
 
@@ -37,7 +38,11 @@ set(CMAKE_CONFIG_PATH ${CMAKE_CURRENT_LIST_DIR}/cmake/configs)
 include(configuration)
 
 add_subdirectory(src)
-add_subdirectory(test)
+
+if(COA_BUILD_TESTS)
+  add_subdirectory(test)
+endif()
+
 add_subdirectory(doc)
 
 include(installation)
-- 
2.45.2

